Sources/Task/TaskCollections.swift

```diff
@@ -25,13 +25,16 @@ extension CollectionType where Generator.Element: FutureType, Generator.Element.
         }
 
         let coalescingDeferred = Deferred<Task<Void>.Result>()
-        let outerProgress = NSProgress(parent: nil, userInfo: nil)
-        outerProgress.totalUnitCount = numericCast(count)
+        let progress = NSProgress(parent: nil, userInfo: nil)
+        progress.totalUnitCount = numericCast(count)
         let group = dispatch_group_create()
 
         for task in self {
-            let innerProgress = NSProgress.wrapped(task, cancellation: nil)
-            outerProgress.adoptChild(innerProgress, orphaned: true, pendingUnitCount: 1)
+            if let task = task as? Task<Generator.Element.Value.Value> {
```

Comment:
Is this basically checking that the value types align? How does that tie into orphaned yea or nay? It would be good to make that connection clear in the source.